src/core/adapters/personalization.ts

```diff
@@ -0,0 +1,126 @@
+import Actions from '../actions';
+import Configuration from '../configuration';
+import ConfigAdapter from './configuration';
+import Selectors from '../selectors';
+import Store from '../store';
+
+namespace Personalization {
+  type ExtractableAction = Actions.SelectRefinement & Actions.AddRefinement;
+
+  export const DAYS_IN_SECONDS = 86400;
+
+  export const extractBias = (action: ExtractableAction, state: Store.State) => {
+    const config = Selectors.config(state).personalization.realTimeBiasing;
+    const byId = Selectors.realTimeBiasesById(state);
+    const { field, value } = extractRefinement(action, state);
+
+    if (!config.attributes[field]) {
+      return null;
+    }
+
+    return {
+      variant: field,
+      key: value,
+      bias: (byId[field] && byId[field][value]) ? {
+        ...byId[field][value],
+        lastUsed: Math.floor(Date.now() / 1000)
+      } : generateNewBias()
+    };
+  };
+
+  // tslint:disable-next-line max-line-length
+  export const extractRefinement = ({ type, payload }: ExtractableAction, state: Store.State): { field: string, value: string } => {
```

Comment:
rather than have this function set `field` and `value` then return them at the end, why not just return them from within your switch statement?

```js
switch (type) {
  case Actions.ADD_REFINEMENT:
    if (payload.range) {
      return { field: payload.navigationId, value: payload.value };
    }
  case Actions.SELECT_REFINEMENT:
    return Selectors.refinementCrumb(state, payload.navigationId, payload.index);
}

return {};
```

Reply:
Fixed.